Fix build

ctjhoa opened the pull request:
https://contractlive.atlassian.net/browse/CPM-13575

Commit:
Remove useless grunt-contrib

Commit:
Bump deps

Commit:
Add package-lock

Commit:
Add grunt-cli

Commit:
Add npm scripts

Commit:
Fix compression

Commit:
Fix uglify task

Commit:
Add CONTRIBUTING documentation

thiberaw reviewed (approved):
Y a plus qu'à tester :sweat_smile:

KamiKillertO committed:
fix all jshint errors

KamiKillertO committed:
Correctly write tests

Use QUnit global and use assert function in tests

KamiKillertO committed:
Add QUnit as dev deps and change server path config

We no longer include a static version of qunit in the project.
And now the webserver only serve the project and not the parent folder

KamiKillertO committed:
Fix indent


Fix indents

KamiKillertO committed:
Split big tests in smaller tests

KamiKillertO committed:
Fix failling tests

Commit:
Add github action
